src/services/WorkspaceFlowRunsApi.ts

```diff
@@ -73,12 +74,28 @@ export class WorkspaceFlowRunsApi extends WorkspaceApi {
     return mapper.map('RunGraphDataResponse', data, 'RunGraphData')
   }
 
-  public async getFlowRunInput(id: string, key: string): Promise<Schema> {
-    const { data } = await this.get<SchemaResponse>(`/${id}/input/${key}`)
+  public async getFlowRunInput(id: string, key: string): Promise<string | Record<string, any> | null> {
+    try {
+      const { data } = await this.get<string | Record<string, any>>(`/${id}/input/${key}`)
+      return data
+    } catch (error) {
+      return null
+    }
+  }
+
+  public async getFlowRunInputDescription(id: string, keyset: FlowRunInputKeyset): Promise <string | null> {
+    return await this.getFlowRunInput(id, keyset.description) as string | null
+  }
 
+  public async getFlowRunInputSchema(id: string, keyset: FlowRunInputKeyset): Promise<Schema> {
+    const data = await this.getFlowRunInput(id, keyset.schema) as SchemaResponse | null
+    if (!data) {
+      throw new Error('Flow run input schema not found')
+    }
     return mapper.map('SchemaResponse', data, 'Schema')
   }
 
+
```

Comment:
So rather than calling the `getFlowRunInput` method and then casting the type I'd just hit the endpoint directly and say what you expect to get back like
```
public async getFlowRunInputDescription(id: string, keyset: FlowRunInputKeyset): Promise<string | null> {
  const { data } = await this.get<string | null>(`/${id}/input/${keyset.description}`)

  return data
}
```
Same for the input schema one. Otherwise we don't really get the type safety we'd like because even if the `getFlowRunInput` type's change these methods won't get type errors.

Reply:
That makes sense, thanks Craig!